R-package/R/lightgbm.R

```diff
@@ -135,6 +149,8 @@ lightgbm <- function(data,
     stop("nrounds should be greater than zero")
   }
 
+  params$num_threads <- nthreads
```

Comment:
```suggestion
  params <- lgb.check.wrapper_param(
    main_param_name = "num_threads"
    , params = params
    , alternative_kwarg_value = num_threads
  )
```

Across LightGBM's Python and R packages, we try to enforce the behavior in `params` overrides LightGBM's defaults set in https://github.com/microsoft/LightGBM/blob/master/include/LightGBM/config.h and anything passed in via keyword arguments.

Please change this line to enforce that behavior, and to ensure that only a single value of `num_threads` and its aliases (https://lightgbm.readthedocs.io/en/latest/Parameters.html#num_threads) is passed down into `lgb.train()`.